src/internal/_reduce.js

```diff
@@ -33,6 +34,20 @@ module.exports = (function() {
         return xf.result(obj.reduce(bind(xf.step, xf), acc));
     }
 
+    function _objectReduce(xf, acc, obj) {
+        var key;
+        for (key in obj) {
```

Comment:
and there should be a test for `obj1 = {a: 1, b: 2}, obj2 = {b: 2, a: 1}` and a non-commutative function and prove that the outputs are the same, e.g. `function(a, b) { return a - b; }`